obal/data/modules/copr_project.py

```diff
@@ -0,0 +1,59 @@
+"""
+Create a Project in Copr
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.copr import copr_cli, CoprCliCommandError, full_name # pylint:disable=import-error,no-name-in-module
+
+
+def main():
+    """
+    Create a Project in Copr
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            user=dict(type='str', required=True),
+            project=dict(type='str', required=True),
+            chroots=dict(type='list', required=True),
+            description=dict(type='str', required=False),
+            unlisted_on_homepage=dict(type='bool', required=False, default=True),
+            delete_after_days=dict(type='str', required=False, default=None),
+        )
+    )
+
+    user = module.params['user']
+    project = module.params['project']
+    chroots = module.params['chroots']
+    description = module.params['description']
+    unlisted_on_homepage = module.params['unlisted_on_homepage']
+    delete_after_days = module.params['delete_after_days']
+
+    if not description:
+        description = project
+
+    command = [
+        'create',
```

Comment:
Take a look at https://github.com/theforeman/obal/pull/344/files#diff-03c091f5d96839468fd435070c76a74c346a0ba1c6f5723d51b4a10c8cb1621a

I was debating how to handle the forking for scratch builds. And this is closely related I think in approach. I considered if I should do a full-scale project module that manages it in all facets or, for the sake of complexity, keeping things separated a bit. Would you do create, edit and fork all in one?

Reply:
Heh, when writing the above comment, I had started another paragraph with "I guess it depends on the workflow we want to implement", but then scraped it…

The way I see this PR is that it mimics what we do today with a combo of [`external_repos.yml` in `foreman-infra`](https://github.com/theforeman/foreman-infra/blob/master/koji/playbooks/external_repos.yml) and [`tool-belt koji`](https://github.com/theforeman/tool_belt/blob/master/lib/tool_belt/koji_tag.rb) aka "manage the base we are using for building". This (for me) includes the ability to change an external repo of a chroot *after* the project has been created (possible with your code, as that is part of the `edit-chroot` code¹) or change the chroots of a project (to add el9, for example, not possible with the current code)

¹: while the edit-chroot code exists and would work, actually executing it would not work, as the module first would try a `copr-cli create` and fail :(

So, TL;DR: no. I think one module should do create/edit (aka: "manage") and another should do fork (aka: "create another one based on a template")